packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts

```diff
@@ -714,6 +716,7 @@ export class BaseTypesVisitor<
   }
 
   UnionTypeDefinition(node: UnionTypeDefinitionNode, key: string | number | undefined, parent: any): string {
+    if (this.config.onlyTypescriptOperationTypes) return '';
     if (this.config.onlyOperationTypes || this.config.onlyEnums) return '';
```

Comment:
I can see `onlyOperationTypes` and `onlyEnums` seems to do the same thing. Could you please help me understand the differences in use cases between the three options? 🙂

Reply:
That is a very valid point... 😅 I think in the future `onlyTypescriptOperationTypes` could actually have the side effect of only emitting the scalars/enums used within the GraphQL operations provided. But since I want to keep that out of scope of this PR, just setting `onlyOperationTypes` and `onlyEnums` to `true` has the same effect.

Reply:
Adjusted in [`bdd83b3` (#10073)](https://github.com/dotansimha/graphql-code-generator/pull/10073/commits/bdd83b330efab9495a17e7cf0a8226c750fc1772)